sdk/go/pulumi/types.go

```diff
@@ -389,11 +391,33 @@ func (o *OutputState) ApplyTWithContext(ctx context.Context, applier interface{}
 		}
 
 		// Fulfill the result.
-		result.fulfillValue(results[0], true, nil)
+		result.fulfillValue(results[0], true, secret, nil)
 	}()
 	return result
 }
 
+// IsSecret returns a bool representing the secretness of the Output
+func (o *OutputState) IsSecret() bool {
+	return o.getState().secret
+}
+
+// SecretT wraps the input in an Output marked as secret. The result remains an output
+// property, and accumulates all implicated dependencies, so that resources can be properly tracked using a DAG.
+// This function does not block awaiting the value; instead, it spawns a Goroutine that will await its availability.
```

Comment:
I don't quite follow the comment above.  What does this function have to do with spawning goroutines?  What does it have to do with a DAG?

Reply:
These comments are copied from Apply. You're right, they don't make much sense here.